apps/examples/testcase/le_tc/kernel/tc_memory_safety.c

```diff
@@ -0,0 +1,408 @@
+	 * Make it a blocking open on the queue, meaning it will block if
+	 * this task tries to read from the queue when the queue is empty
+	 *
+	 *   O_CREAT - the queue will get created if it does not already exist.
+	 *   O_RDONLY - we are only planning to read from the queue.
+	 *
+	 * Open the queue, and create it if the sending process hasn't
+	 * already created it.
+	 */
+
+	g_recv_mqfd = mq_open("mqueue", O_RDONLY | O_CREAT, 0666, &attr);
+	if (g_recv_mqfd == (mqd_t)ERROR) {
+		printf("tc_mqueue_mq_open FAIL\n");
+		pthread_exit((pthread_addr_t)1);
+	}
+
+	/* Perform the receive TEST_RECEIVE_NMSGS times */
+
+	for (msg_idx = 0; msg_idx < TEST_RECEIVE_NMSGS; msg_idx++) {
+		memset(msg_buffer, 0xaa, TEST_MSGLEN);
+		nbytes = mq_receive(g_recv_mqfd, msg_buffer, TEST_MSGLEN, 0);
+		if (nbytes < 0) {
+			/* mq_receive failed.  If the error is because of EINTR then
+			 * it is not a failure.
+			 */
+
+			if (errno != EINTR) {
+				printf("tc_mqueue_mq_receive FAIL : failure on msg %d, errno=%d\n", msg_idx, errno);
+				nerrors++;
+			} else {
+				tckndbg("mq_receive interrupted!\n");
+			}
+		} else if (nbytes != TEST_MSGLEN) {
+			printf("tc_mqueue_mq_receive FAIL : return bad size %d on msg %d\n", nbytes, msg_idx);
+			nerrors++;
+		} else if (memcmp(TEST_MESSAGE, msg_buffer, nbytes) != OK) {
+			int msg_chk;
+
+			printf("tc_mqueue_mq_receive FAIL : returned corrupt message on msg %d\n", msg_idx);
+			tckndbg("receiver_thread:                  i  Expected Received\n");
+
+			for (msg_chk = 0; msg_chk < TEST_MSGLEN - 1; msg_chk++) {
+				if (isprint(msg_buffer[msg_chk])) {
+					tckndbg("receiver_thread: %2d %02x (%c) %02x (%c)\n", msg_chk, TEST_MESSAGE[msg_chk], TEST_MESSAGE[msg_chk], msg_buffer[msg_chk], msg_buffer[msg_chk]);
+				} else {
+					tckndbg("receiver_thread: %2d %02x (%c) %02x\n", msg_chk, TEST_MESSAGE[msg_chk], TEST_MESSAGE[msg_chk], msg_buffer[msg_chk]);
+				}
+			}
+
+			tckndbg("receiver_thread: %2d 00 %02x\n", msg_chk, msg_buffer[msg_chk]);
+		}
+	}
+
+	/* Close the queue and return success */
+
+	if (mq_close(g_recv_mqfd) < 0) {
+		printf("tc_mqueue_mq_close FAIL\n");
+		nerrors++;
+	} else {
+		g_recv_mqfd = NULL;
+	}
+
+	pthread_exit((pthread_addr_t)nerrors);
+	return (pthread_addr_t)nerrors;
+}
+
+/**
+* @fn                   :infiniteloop
+* @description          :Function for tc_memory_safety
+* @return               :void*
+*/
+static void *infiniteloop(void *arg)
+{
+	int i = 0;
+	while (1) {
+		i++;
+		i = i % 1000;
+		if (g_exit == 1) {
+			break;
+		}
+	}
+	return NULL;
+}
+
+/**
+ * @fn                  :tc_memory_sefety_with_mqueue
+ * @brief               :Tests mqueue APIs open, close, send, receive.
+ * @Scenario            :Tests mqueue APIs open, close, send, receive.
+ * API's covered        :mq_open, mq_close, mq_send, mq_receive
+ * Preconditions        :none
+ * Postconditions       :none
+ * @return              :void
+ */
+static void tc_memory_sefety_with_mqueue(void)
```

Comment:
same as above(should change sefety to safety.)

Reply:
fixed